### DISCUSSION TITLE
Budibase Native DB - Limits

### DISCUSSION THREAD

**Discussion by scotttiey** [Budibase/budibase#12414] [Help] [answered]
Hi I can't find any info on things like row limits etc for the Budibase native db. We would like to use this and have everything managed in one place (no external db setup) but we want to make sure it can scale. Thanks Scott

**Comment by ConorWebb96** [comment]
Hey @scotttiey,

Are you looking into self-hosting or using cloud?

Cloud has a few more limits on it. Rows are capped to 2000 rows, you can only run 200 automations a month and you can only install 10 custom plugins at a time on the free tier.
<img width="605" alt="Screenshot 2023-11-22 at 09 15 35" src="https://github.com/Budibase/budibase/assets/126772285/2e5f5a4e-a19b-40d7-9f74-75ae942bb16d">

Self-host doesn't have the row limitation or the automation limits. However it will have a user limitation whenever the new pricing plan is implemented. It also has the custom plugin limitation too.

I hope this helps.

**Comment by scotttiey** [comment]
Hi Conor

It was using the Cloud version with the new pricing plan (not the free plan) but utilising the built in Couch DB not Self Hosting.

Are there row/storage/automation limits with that?

Kind Regards

Scott Miller

Cloud 9 Assist | tel: +44 203 667 2777 | web: www.cloud9assist.com<http://www.cloud9assist.com/>


From: Conor Webb ***@***.***>
Date: Wednesday, 22 November 2023 at 09:24
To: Budibase/budibase ***@***.***>
Cc: Scott Miller ***@***.***>, Mention ***@***.***>
Subject: Re: [Budibase/budibase] Budibase Native DB - Limits (Discussion #12414)

Hey @scotttiey<https://github.com/scotttiey>,

Are you looking into self-hosting or using cloud?

Cloud has a few more limits on it. Rows are capped to 2000 rows, you can only run 200 automations a month and you can only install 10 custom plugins at a time on the free tier.
[Screenshot 2023-11-22 at 09 15 35]<https://user-images.githubusercontent.com/126772285/284851133-2e5f5a4e-a19b-40d7-9f74-75ae942bb16d.png>

Self-host doesn't have the row limitation or the automation limits. However it will have a user limitation whenever the new pricing plan is implemented. It also has the custom plugin limitation too.

I hope this helps.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @scotttiey,

Apologies, I don't believe you'll have row limits if on one of the new plans. You can easily check by going to to your account area in your Budibase instance and see what you have access to. I've attached a screenshot of what I'm talking about below.

<img width="975" alt="Screenshot 2023-11-22 at 10 19 00" src="https://github.com/Budibase/budibase/assets/126772285/ee0917f9-b4f3-4354-8855-a43602b862dc">

I hope this makes sense!

**Comment by scotttiey** [comment]
Cheers Conor. Thank you for your help

From: Conor Webb ***@***.***>
Date: Wednesday, 22 November 2023 at 12:39
To: Budibase/budibase ***@***.***>
Cc: Scott Miller ***@***.***>, Mention ***@***.***>
Subject: Re: [Budibase/budibase] Budibase Native DB - Limits (Discussion #12414)

Hey @scotttiey<https://github.com/scotttiey>,

Apologies, I don't believe you'll have row limits if on one of the new plans. You can easily check by going to to your account area in your Budibase instance and see what you have access to. I've attached a screenshot of what I'm talking about below.
[Screenshot 2023-11-22 at 10 19 00]<https://user-images.githubusercontent.com/126772285/284905363-ee0917f9-b4f3-4354-8855-a43602b862dc.png>

I hope this makes sense!
